execution_engine/src/core/runtime/standard_payment_internal.rs

```diff
@@ -50,7 +50,13 @@ where
         target: URef,
         amount: U512,
     ) -> Result<(), ApiError> {
-        let mint_contract_hash = self.get_mint_contract();
+        let mint_contract_hash = match self.get_mint_contract() {
+            Ok(mint_hash) => mint_hash,
+            Err(exec_error) => {
+                let maybe_api_error: Option<ApiError> = exec_error.into();
```

Comment:
This will always be `None` I believe - please double check.

Seems like this probably warrants a new `ApiError` variant: `MissingSystemContractHash` with doc comment making it clear this is a bug if returned.